specification.md

```diff
@@ -1642,6 +1647,48 @@ In this case the `failOnValidationErrors` default value of `true` is assumed.
 The `dataInputSchema` property validates the [workflow data input](#Workflow-Data-Input). In case of 
 a starting [Event state](#Event-state), it is not used to validate its event payloads.
 
+The `secrets` property allows you to use sensitive information such as passwords, OAuth tokens, ssh keys, etc. inside your
+Workflow expressions. 
+
+It has two possible types, `string` or `array`. 
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an array of names of the secrets, for example:
+
+```json
+"secrets": "file://workflowsecrets.json"
+```
+
+If `array` type, it defines an array (of string types) which contains the names of the secrets, for example:
+
+```json
+"secrets": ["MY_PASSWORD", "MY_STORAGE_KEY", "MY_ACCOUNT"]
+```
+
+For more information about Workflow secrets, reference the [Workflow Secrets section](#Workflow-Secrets).
+
+The `globals` property can be used to define Workflow global values
+which are accessible in [Workflow Expressions](#Workflow-Expressions).
+
+It has two possible types, `string` or `object`.
+If `string` type, it is an URI pointing to a JSON or YAML document
+which contains an object of global definitions, for example:
```

Comment:
Not saying I see a purpose to explicitly support this, but technically could the file contain just any valid json/yaml?  I think object is the most desirable use-case, but if the file contains an array, it seems like that'd be valid to just reference that data via `$GLOBALS[0]`, etc?

Should that be supported and described?  Or should we say that a runtime should consider that an error if the top-level content of the file is anything but an 'object'?

Reply:
good point. up to you guys really. since globals can also be defined in-line we should probably restrict its type however. 
we can keep it as object type (which means the file should be a json object) or we can expand it to be object or array types (which then would restrict the read in files to that as well). 

either way, if the globals are read from file, they can only contain json which is also possible to define inline in the workflow definition